assert/assertions_test.go

```diff
@@ -208,6 +208,12 @@ func TestEqual(t *testing.T) {
 	if !Equal(mockT, &struct{}{}, &struct{}{}) {
 		t.Error("Equal should return true (pointer equality is based on equality of underlying value)")
 	}
+	if Equal(mockT, pointerOf("a"), pointerOf("b")) {
+		t.Error("Equal should return false (pointer equality is based on equality of underlying value)")
+	}
+	if Equal(mockT, "a", pointerOf("b")) {
+		t.Error("Equal should return false")
+	}
```

Comment:
Would these tests have failed before your changes, though? It doesn't seem that way. Could you verify?